### PR TITLE
Removed type info from serializer methods

### PR DESCRIPTION
**Pull request by tejainece**

### CODE REVIEW

File: example/basic/basic_main.dart [hunk 1]
```diff
@@ -4,17 +4,23 @@ import 'package:jaguar_serializer/serializer.dart';
 
 part 'basic_main.g.dart';
 
-@GenSerializer()
+@GenSerializer(addTypeInfo: true)
```
Comment (lejard-h) [inline example/basic/basic_main.dart:7]: So it means that, the user don't have choice at runtime now ? or he needs to create 2 serializer ?

File: lib/src/serializer/repo.dart [hunk 2]
```diff
@@ -81,30 +81,18 @@ class SerializerRepo {
    *
    * The [typeInfoKey] can be override using the [useTypeInfoKey] option.
    */
-  dynamic to(dynamic object,
-      {bool withTypeInfo: false, String useTypeInfoKey}) {
-    useTypeInfoKey ??= typeInfoKey;
+  dynamic to(dynamic object) {
```
Comment (lejard-h) [inline lib/src/serializer/repo.dart]: not possible to change the typeKey anymore ?

File: lib/src/json/json.dart [hunk 3]
```diff
@@ -39,6 +39,5 @@ class JsonRepo extends SerializerRepo {
   @override
   dynamic to(dynamic object,
```
Comment (lejard-h) [inline lib/src/json/json.dart]: should have the same api as SerializerRepo.to

File: example/basic/basic_main.dart (hunk 1 above)
Comment (tejainece) [reply example/basic/basic_main.dart:7]: Yes. Why would you need two different type keys?

Comment (lejard-h) [reply example/basic/basic_main.dart:7]: with you proposition, after generation, one have no choice to embeded or not the typeInfo, that is why I think it is better to keep that option at runtime, `Serializer.encode(object, withTypeInfo: true);`

Comment (lejard-h) [reply example/basic/basic_main.dart:7]: @tejainece Can you change this ?

**Pull request closed without merging**